src/integrations/admin/activation-cleanup-integration.php

```diff
@@ -49,6 +60,9 @@ public function register_hooks() {
 	public function register_cleanup_routine() {
 		$first_activated_on = $this->options_helper->get( 'first_activated_on', false );
 
+		if ( ! $this->indexable_helper->should_index_indexables() ) {
+			return;
+		}
```

Comment:
Wouldn't it be more efficient if you would check this before fetching the option above?

Reply:
Done!